### PR TITLE
stm32_common/flashpage: fix stm32l4 erase error

### PR DESCRIPTION
**Pull request by fjmolinas**
### Contribution description

This PR fixes a ~couple of ~bug~s~ in flashpage stm32l4 implementation.

- Fixes `_erase()` for stm32l4. When erasing the page number to write was not reset before `oring`
the new one, this resulted in the wrong page being erase after a couple of erase sequence on different pages. 

~~- In `flash_common` `FLASH_SR_EOP` is cleared by writing to 0, when RM states that it is cleared by writing 1 (check any STM32 `FLASH_SR` register description).~~
~~- When writing to flash the first operation is performed before checking if there was any ongoing operation. This should also be done according to the RM.~~

### Testing procedure

~~The second and third point don't need testing, just verify my statement in the RM.~~

For the first point in master, for any STML4:

`make -C tests/periph_flashpage/ BOARD=nucleo-l476rg flash -j3 term`

```
> test 128
2019-07-03 09:49:06,670 - INFO #  test 128
2019-07-03 09:49:06,719 - INFO # wrote local page buffer to flash page 128 at addr 0x8040000
> test 64
2019-07-03 09:49:07,813 - INFO #  test 64
2019-07-03 09:49:07,839 - INFO # error verifying the content of page 64
```

With this PR:

```
> test 128
2019-07-03 09:55:58,557 - INFO #  test 128
2019-07-03 09:55:58,606 - INFO # wrote local page buffer to flash page 128 at addr 0x8040000
> test 64
2019-07-03 09:55:59,893 - INFO #  test 64
2019-07-03 09:55:59,942 - INFO # wrote local page buffer to flash page 64 at addr 0x8020000
```

### CODE REVIEW

File: cpu/stm32_common/periph/flash_common.c [hunk 1]
```diff
@@ -67,6 +67,7 @@ void _wait_for_pending_operations(void)
 
     /* Clear 'end of operation' bit in status register */
     if (FLASH->SR & FLASH_SR_EOP) {
-        FLASH->SR &= ~(FLASH_SR_EOP);
+        FLASH->SR |= FLASH_SR_EOP;
```
Comment (aabadie) [inline cpu/stm32_common/periph/flash_common.c]: Maybe we could just remove the if line above and just keep this line?

File: cpu/stm32_common/periph/flash_common.c [hunk 2]
```diff
@@ -67,6 +67,7 @@ void _wait_for_pending_operations(void)
...
     }
+
```
Comment (aabadie) [inline cpu/stm32_common/periph/flash_common.c]: unrelated

**Comment by fjmolinas** [comment]
> Some changes in this PR are affecting all STMs that have flashpage support, so need to be tested (not only stm32l4 as mentioned in the PR title)

Yes and no:

**Comment by fjmolinas** [comment]
> Some changes in this PR are affecting all STMs that have flashpage support, so need to be tested (not only stm32l4 as mentioned in the PR title)

Sorry I had looked only at some datasheets and for STM32F2, STM32F4, STM32F7 & STM32L4:

> Bit 0 EOP: End of operation
> Set by hardware when one or more Flash memory operations (program/erase) has/have
> completed successfully. It is set only if the end of operation interrupts are enabled (EOPIE = 1).
> Cleared by writing a 1.

So it should be tested for: STM32F1, STM32F3, STM32L0, STM32L1.

File: cpu/stm32_common/periph/flash_common.c (hunk 1 above)
Comment (fjmolinas) [reply cpu/stm32_common/periph/flash_common.c]: We could do this and guard it for STM32F2, STM32F4, STM32F7 & STM32L4 since EOPIE is not set this is not needed for these platforms, a proper commit comment would justify and documment in case in a future EOPIE are used. What do you think?
